### PR TITLE
Add daa clickhouse

### PR DESCRIPTION
**Pull request by tspenov**
Add DAA from clickhouse and realtime DAA

### CODE REVIEW

File: lib/sanbase/clickhouse/erc20_daily_active_addresses.ex [hunk 1]
```diff
@@ -1,22 +1,151 @@
...
+  Returns a list of tupples {contract, active_addresses}
   """
 
-  use Ecto.Schema
+  @spec average_active_addresses(
+          contracts,
+          %DateTime{},
+          %DateTime{}
+        ) :: {:ok, list(contract_daa_tupple)} | {:error, String.t()}
+
```
Comment (IvanIvanoff) [inline lib/sanbase/clickhouse/erc20_daily_active_addresses.ex]: there's one line

File: lib/sanbase/clickhouse/erc20_daily_active_addresses.ex [hunk 2]
```diff
@@ -1,22 +1,151 @@
...
+      SELECT contract, dt, (toFloat64(uniq(address)) * toFloat64(#{today_coefficient})) as total_addresses
+      FROM erc20_daily_active_addresses_list
+      WHERE contract IN (?3) and dt >= toDateTime(today()) AND
+      dt >= toDateTime(?1) AND
+      dt <= toDateTime(?2)
+      GROUP BY contract, dt
+    )
+    GROUP BY contract
+    ORDER BY contract
```
Comment (IvanIvanoff) [inline lib/sanbase/clickhouse/erc20_daily_active_addresses.ex]: do we need order by?

File: lib/sanbase/clickhouse/erc20_daily_active_addresses.ex [hunk 3]
```diff
@@ -1,22 +1,151 @@
...
+      WHERE
+      contract IN (?3) AND
+      dt < toDateTime(today()) AND
+      dt >= toDateTime(?1) AND
+      dt <= toDateTime(?2)
+      GROUP BY contract, dt
+
+      UNION ALL
+
+      SELECT contract, dt, (toFloat64(uniq(address)) * toFloat64(#{today_coefficient})) as total_addresses
```
Comment (IvanIvanoff) [inline lib/sanbase/clickhouse/erc20_daily_active_addresses.ex]: should we use something like `toStartOfDay(dt) as dt`?

File: lib/sanbase/clickhouse/erc20_daily_active_addresses.ex [hunk 4]
```diff
@@ -1,22 +1,151 @@
...
+  Returns the active addresses for a contract chunked in intervals between [from, to]
+  If last day is included in the [from, to] the value is the realtime value in the current moment
+  """
+  @spec average_active_addresses(
+          String.t(),
+          %DateTime{},
+          %DateTime{},
+          String.t()
+        ) :: {:ok, list(active_addresses)} | {:error, String.t()}
+
```
Comment (IvanIvanoff) [inline lib/sanbase/clickhouse/erc20_daily_active_addresses.ex]: empty line

File: lib/sanbase/clickhouse/eth_daily_active_addresses.ex [hunk 5]
```diff
@@ -1,21 +1,138 @@
...
+      UNION ALL
+      
+      SELECT dt, (toFloat64(uniq(address)) * toFloat64(#{today_coefficient})) as total_addresses
+      FROM eth_daily_active_addresses_list
+      WHERE dt >= toDateTime(today()) AND
+      dt >= toDateTime(?1) AND
+      dt <= toDateTime(?2)
+      GROUP BY dt
+    )
+    """
```
Comment (IvanIvanoff) [inline lib/sanbase/clickhouse/eth_daily_active_addresses.ex]: here we miss the filling with zeroes, is this an issue?

Comment (tspenov) [reply lib/sanbase/clickhouse/eth_daily_active_addresses.ex]: * for bucketing we will show`0` for the intervals that have that 
* for calc AVG `NULL`s will be excluded in the calculation - so 
examples: (on `2015-07-29 00:00:00Z` e don't have DAA for ETH)
```
{:ok,
 [
   %{active_addresses: 0, datetime: #DateTime<2015-07-29 00:00:00Z>},
   %{active_addresses: 9207, datetime: #DateTime<2015-07-30 00:00:00Z>}
 ]}
```

```
{:ok, 9207}
```

Comment (tspenov) [reply lib/sanbase/clickhouse/eth_daily_active_addresses.ex]: Using Timescale returns the same result: 
```
{:ok,
 [
   %{active_addresses: 0, datetime: #DateTime<2015-07-28 00:00:00Z>},
   %{active_addresses: 0, datetime: #DateTime<2015-07-29 00:00:00Z>},
   %{active_addresses: 9207, datetime: #DateTime<2015-07-30 00:00:00Z>}
 ]}
iex(45)> DailyActiveAddresses.average_active_addresses("ETH", days_ago.(1261), days_ago.(1259))
[16:18:51.443][debug][] QUERY OK db=128.5ms queue=0.1ms
SELECT contract_address, coalesce(avg(active_addresses), 0) AS value
FROM etherbi.eth_daily_active_addresses
WHERE timestamp >= $1 AND timestamp <= $2 AND contract_address = ANY($3)
GROUP BY contract_address
 [{{2015, 7, 28}, {14, 18, 51, 314260}}, {{2015, 7, 30}, {14, 18, 51, 314482}}, ["ETH"]]
{:ok, [{"ETH", 9207}]}
```

**Pull request merged**